tasklist/common/src/main/java/io/camunda/tasklist/es/ElasticsearchConnector.java

```diff
@@ -164,27 +172,75 @@ public RestHighLevelClient createEsClient(ElasticsearchProperties elsConfig) {
   }
 
   private HttpAsyncClientBuilder configureHttpClient(
-      HttpAsyncClientBuilder httpAsyncClientBuilder, ElasticsearchProperties elsConfig) {
+      final HttpAsyncClientBuilder httpAsyncClientBuilder,
+      final ElasticsearchProperties elsConfig) {
     setupAuthentication(httpAsyncClientBuilder, elsConfig);
+
+    LOGGER.trace("Attempt to load interceptor plugins");
+    if (elsConfig.getInterceptorPlugins() != null) {
+      loadInterceptorPlugins(httpAsyncClientBuilder, elsConfig);
+    }
+
     if (elsConfig.getSsl() != null) {
       setupSSLContext(httpAsyncClientBuilder, elsConfig.getSsl());
     }
     return httpAsyncClientBuilder;
   }
 
+  private void loadInterceptorPlugins(
+      final HttpAsyncClientBuilder httpAsyncClientBuilder,
+      final ElasticsearchProperties elsConfig) {
+    LOGGER.trace("Plugins detected to be not empty {}", elsConfig.getInterceptorPlugins());
+    final List<InterceptorPluginProperties> interceptors = elsConfig.getInterceptorPlugins();
+    for (final InterceptorPluginProperties interceptor : interceptors) {
+      LOGGER.trace("Attempting to register {}", interceptor.getId());
+      try {
+        // WARNING! Due to the nature of interceptors, by the moment they (interceptors)
+        // are executed, the below class loader will close the JAR file hence
+        // to avoid NoClassDefFoundError we must not close this class loader.
+        final var classLoader = ExternalJarClassLoader.ofPath(Paths.get(interceptor.getJarPath()));
+
+        final var pluginClass = classLoader.loadClass(interceptor.getClassName());
+        final var plugin = ReflectUtil.newInstance(pluginClass);
+
+        if (plugin instanceof final DatabaseCustomHeaderSupplier dchs) {
+          LOGGER.trace(
+              "Plugin {} appears to be a DB Header Provider. Registering with interceptor",
+              interceptor.getId());
+          httpAsyncClientBuilder.addInterceptorLast(
+              (HttpRequestInterceptor)
+                  (httpRequest, httpContext) -> {
+                    try {
+                      final var customHeader =
+                          ThreadContextUtil.callWithClassLoader(
+                              dchs::getElasticsearchCustomHeader, classLoader);
+                      httpRequest.addHeader(customHeader.key(), customHeader.value());
+                    } catch (final Exception e) {
+                      throw new RuntimeException(e);
+                    }
```

Comment:
:wrench: I think this is because `ThreadContextUtil.callWithClassLoader` throws an exception in its signature, so you're forced to wrap it here, even though your plugin does not throw. How about we add a new `ThreadContextUtil.supplyWithClassLoader` which takes a `Supplier<V>` instead of a `Callable<V>`? I'm sure there would be other usages elsewhere too.